pkg/operator/ceph/object/objectstore.go

```diff
@@ -872,139 +847,226 @@ func sharedPoolsExist(objContext *Context, sharedPools cephv1.ObjectSharedPoolsS
+
+	for _, pp := range sharedPools.PoolPlacements {
+		if _, ok := existing[pp.MetadataPoolName]; !ok {
+			return fmt.Errorf("sharedPool do not exist: pool %s for placement %s", pp.MetadataPoolName, pp.Name)
+		}
+		if _, ok := existing[pp.DataPoolName]; !ok {
+			return fmt.Errorf("sharedPool do not exist: pool %s for placement %s", pp.DataPoolName, pp.Name)
+		}
+		if _, ok := existing[pp.DataNonECPoolName]; !ok {
+			return fmt.Errorf("sharedPool do not exist: pool %s for placement %s", pp.DataNonECPoolName, pp.Name)
+		}
+		for _, sc := range pp.StorageClasses {
+			if _, ok := existing[sc.DataPoolName]; !ok {
+				return fmt.Errorf("sharedPool do not exist: pool %s for StorageClass %s", sc.DataPoolName, sc.Name)
+			}
+		}
 	}
 
-	logger.Info("verified shared pools exist")
 	return nil
 }
 
-func applyExpectedRadosNamespaceSettings(zoneConfig map[string]interface{}, metadataPrefix, dataPrefix, dataPool string) error {
-	// Update the necessary fields for RAODS namespaces
-	zoneConfig["domain_root"] = metadataPrefix + "meta.root"
-	zoneConfig["control_pool"] = metadataPrefix + "control"
-	zoneConfig["gc_pool"] = metadataPrefix + "log.gc"
-	zoneConfig["lc_pool"] = metadataPrefix + "log.lc"
-	zoneConfig["log_pool"] = metadataPrefix + "log"
-	zoneConfig["intent_log_pool"] = metadataPrefix + "log.intent"
-	zoneConfig["usage_log_pool"] = metadataPrefix + "log.usage"
-	zoneConfig["roles_pool"] = metadataPrefix + "meta.roles"
-	zoneConfig["reshard_pool"] = metadataPrefix + "log.reshard"
-	zoneConfig["user_keys_pool"] = metadataPrefix + "meta.users.keys"
-	zoneConfig["user_email_pool"] = metadataPrefix + "meta.users.email"
-	zoneConfig["user_swift_pool"] = metadataPrefix + "meta.users.swift"
-	zoneConfig["user_uid_pool"] = metadataPrefix + "meta.users.uid"
-	zoneConfig["otp_pool"] = metadataPrefix + "otp"
-	zoneConfig["notif_pool"] = metadataPrefix + "log.notif"
-
-	placementPools, ok := zoneConfig["placement_pools"].([]interface{})
-	if !ok {
-		return fmt.Errorf("failed to parse placement_pools")
+func adjustZoneDefaultPools(zone map[string]interface{}, spec cephv1.ObjectSharedPoolsSpec) (map[string]interface{}, error) {
+	name, err := getObjProperty[string](zone, "name")
+	if err != nil {
+		return nil, fmt.Errorf("unable to get zone name: %w", err)
 	}
-	if len(placementPools) == 0 {
-		return fmt.Errorf("no placement pools")
+
+	zone, err = deepCopyJson(zone)
+	if err != nil {
+		return nil, fmt.Errorf("unable to deep copy zone %s: %w", name, err)
+	}
+
+	defaultMetaPool := getDefaultMetadataPool(spec)
+	if defaultMetaPool == "" {
+		// default pool is not presented in shared pool spec
+		return zone, nil
+	}
+	// add zone namespace to metadata pool to safely share accorss rgw instances or zones.
+	// in non-multisite case zone name equals to rgw instance name
+	defaultMetaPool = defaultMetaPool + ":" + name
+	zonePoolNSSuffix := map[string]string{
+		"domain_root":     ".meta.pool",
```

Comment:
This looks like it changed above from "meta.root", was that intentional?
```suggestion
		"domain_root":     ".meta.root",
```